components/generic/ContentCard.vue

```diff
@@ -256,6 +257,33 @@
 
       imageNotFound() {
         this.cardImageUrl = '';
+      },
```

Comment:
This introduces a lot of logic to the component.

Let's instead move the two buttons and all of this accompanying interaction logic into a new component.